src/axom/core/Macros.hpp

```diff
@@ -257,4 +255,38 @@
   className(className&&) = delete;             \
   className& operator=(className&&) = delete
 
-#endif /* AXOM_MACROS_HPP_ */
+/*!
+ * \def AXOM_TYPED_TEST(CaseName, TestName)
+ * \brief Minor tweak of gtest's TYPED_TEST macro to work with device code
+ * \note Can be used after including `gtest/gtest.h`
+ */
+// Specifically, we expose the `TestBody()` method as public instead of private
+#define AXOM_TYPED_TEST(CaseName, TestName)                                         \
+  static_assert(sizeof(GTEST_STRINGIFY_(TestName)) > 1,                             \
+                "test-name must not be empty");                                     \
+  template <typename gtest_TypeParam_>                                              \
+  class GTEST_TEST_CLASS_NAME_(CaseName, TestName)                                  \
+    : public CaseName<gtest_TypeParam_>                                             \
+  {                                                                                 \
+  public:                                                                           \
```

Comment:
This macro is a copy of the `TYPED_TEST` macro from gtest.
The only difference was changing this line from `private' to `public` to allow it to work with CUDA device kernels.

Without this change, I got the following error:
> The enclosing parent function ("TestBody") for an extended __host__ __device__ lambda cannot have private or protected access within its class

Reply:
Thanks @joshessman-llnl -- I thought about  both of these.
We don't currently have another place to put test-specific utility code. 
If I placed it somewhere else, we'd need to incorporate that into the build system, so I figured this was the easiest place for now.

Reply:
I also thought about patching this line in blt's gtest, but that would affect all blt users, which seemed like a bigger decision.